src/tracks/playabletrack/wavetrack/ui/WaveTrackView.cpp

```diff
@@ -703,6 +704,46 @@ std::pair<
          mCloseHandle,
          *pWaveTrackView, *this, state ) )
          results.second.push_back( pHandle );
+
+      auto channels = TrackList::Channels(wt.get());
+      if(channels.size() > 1) {
+         // Only one cell is tested and we need to know
+         // which one and it's relative location to the border.
+         auto subviews = pWaveTrackView->GetVisibleSubviews();
+         auto currentSubview = std::find(subviews.begin(), subviews.end(), shared_from_this());
+         if (currentSubview != subviews.end())
+         {
+            auto currentSubviewIndex = std::distance(subviews.begin(), currentSubview);
+            
+            const auto py = state.state.GetY();
+            const auto topBorderHit = std::abs(py - state.rect.GetTop())
+               <= WaveTrackView::kChannelSeparatorThickness / 2;
+            const auto bottomBorderHit = std::abs(py - state.rect.GetBottom())
+               <= WaveTrackView::kChannelSeparatorThickness / 2;
+
+            auto currentChannel = channels.find(wt.get());
+            auto currentChannelIndex = std::distance(channels.begin(), currentChannel);
+
+            if (//for not-last-view check the bottom border hit
+               ((currentChannelIndex != channels.size() - 1)
+                  && (currentSubviewIndex == static_cast<int>(subviews.size()) - 1)
+                  && bottomBorderHit)
+               ||
+               //or for not-first-view check the top border hit
+               ((currentChannelIndex != 0) && currentSubviewIndex == 0 && topBorderHit))
+            {
+               //depending on which border hit test succeeded on we
+               //need to choose a proper target for resizing
+               auto it = bottomBorderHit ? currentChannel : currentChannel.advance(-1);
+               auto result = std::static_pointer_cast<UIHandle>(
+                  std::make_shared<TrackPanelResizeHandle>((*it)->shared_from_this(), py)
+               );
+               result = AssignUIHandlePtr(mResizeHandle, result);
+               results.second.push_back(result);
+            }
+         }
+      }
+
```

Comment:
Wow, I'm glad you figured all this out.  But it makes me wonder if there is a simpler way to do this.

Should CellularPanel support a notion of cells with hit zones that extend outside their visible areas?  Could the resizer cell make lower-priority hits that simply combine with the rest of the WaveTrackSubView's hits?  Would that have simplified this project? I need to think about it for the future.

Reply:
And thank you for keeping it generalized, not assuming at most two channels.